### ISSUE TITLE
How to use a PJRT plugin from Pytorch?

### ISSUE THREAD

**Issue by wwtghx** [openxla/xla#7855]
In the file https://github.com/openxla/xla/blob/main/xla/pjrt/c/docs/pjrt_integration_guide.md,  "How to use a PJRT plugin from JAX" is found, but I want to know "How to use a PJRT plugin from Pytorch"?

**Comment by cheshire** [comment]
I think the expected usage pattern is via https://github.com/pytorch/xla, which itself should use the PJRT API.

**Comment by cheshire** [comment]
@JackCaoG for further comments

**Comment by JackCaoG** [comment]
Depending on what you mean by `use`. If you are a user who wants to use PyTorch/XLA with PJRT, it is already the default runtime after 2.1 release. Check out our doc at  https://github.com/pytorch/xla/blob/master/docs/pjrt.md#pjrt-runtime. 

If you are a backend vender who want to integrate PyTorch/XLA with your backend. @will-cromar recently started the effort of developing the python PJRT plugin for PyTorch/XLA. My understanding is that registration process will be different, but you would also need to implement the PJRT CAPI similar to https://github.com/openxla/xla/blob/main/xla/pjrt/c/docs/pjrt_integration_guide.md#step-1-implement-pjrt-c-api-interface .

**Comment by wwtghx** [comment]
Yes, I am a backend vender who want to integrate PyTorch/XLA with my backend.

**Comment by JackCaoG** [comment]
Great, we have helped a couple backend vender to integrated into pytorch/xla. @will-cromar recently started the python plugin effort in https://github.com/pytorch/xla/pull/5644, I expect it to be more mature in a couple months. In the mean time I think you can start with implement the PJRT CAPIs, after that we can help with the backend registration logic.

**Comment by will-cromar** [comment]
Hey, @wwtghx. It's possible today to use a PJRT Plugin from PyTorch/XLA, but the process is rough around the edges. It basically amounts to updating lists and special cases throughout this repository as in https://github.com/pytorch/xla/pull/5428.

This is an active area of work, and the end goal is to simplify the process and implement a registration mechanism for external plugins like JAX has. I started https://github.com/pytorch/xla/issues/6242 to collect my ideas and track progress. Please take a look and let me know if you have any feedback on the design.
